src/tree/Element.mjs

```diff
@@ -112,6 +112,12 @@ export default class Element {
         this._w = 0;
 
         this._h = 0;
+
+        /**
+         * When enablePointer setting is on, this flag will be checked if the element should respond
+         * to the mouse events.
+         */
+        this._collision = true;
```

Comment:
Apologies for barging in on this, but semantically, wouldn't it make more sense for the flag to represent that a certain element was "transparent", collision-wise?

Also, creating a large number of elements is something that is already costly on some very resource-limited devices - do we really need to initialise another flag for something that will probably (and this is me assuming stuff) be the exception, or not used at all in projects that don't need to support the LG Wand?

Reply:
Hi @sky-pedrofernandes

Can you elaborate on "transparency" a little bit more?

While it's true that the `_collision` flag could be hidden behind the setting option flag for idiomatic reasons, the memory footprint it leaves is negligible not only on its own but also compared to all other memory overhead present to even take into consideration. When that becomes necessary, proper specific optimizations for RAM usage can be applied instead that will have a significant impact.

Reply:
I've been wrapping my head around the best way to explain my point of view and apologise upfront should I fail miserably.

The way it stands, right now, with the flag having the default value of `collision: true` what we're saying is that **all elements** will be eligible for collision-detection unless you specify otherwise, correct? To put it another way, what's being provided to the Lightning developer is the ability to say which elements *should not be taken into consideration*.
In this situation what we'll effectively be doing is then specifying which elements should be `transparent` during collision-testing, so perhaps the flag could reflect that explicitly. As I mentioned above, it's a simple semantic nuance and something that, personally, made sense to me.

Now, thinking about this a bit more, if we analyse it exclusively from a performance point of view, I would argue that the default value should actually be `collision: false`, so that you would really only collision-test elements you **knew were going to allow interactions with the pointer**.
I do understand that this would make it a bit harder to use as you'd need to explicitly mark the elements you wanted to consider for pointer interaction, but it would avoid a lot of collision checks.
In this case, I think the `collision` flag would actually be much more appropriate than `transparency`.

Finally, regarding the memory footprint - I'm not saying that this would be the biggest of our problems, and you can always argue (correctly, I might add) that the effect will be mostly negligible. But it won't be 0. I do apologise for being particularly picky with this, but some of the devices we're working with are really limited in terms of performance and anything that could be made to avoid impacting that even further, would be most welcome - even more so because those devices won't actually support any pointer devices.

Reply:
@sky-pedrofernandes

That's a very good point, indeed having all the elements eligible for collision by default is not ideal (although they are initially narrowed down to all active elements). I reversed the logic but as a consequence added a branch marking mechanism to handle nested cases as with current implementation this would otherwise put stress on the developer to mark the entire branch and not just the final element manually. I might revisit the entire implementation to find a more elegant solution.